extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -178,11 +221,53 @@ export class AmpAutocomplete extends AMP.BaseElement {
     this.action_ = Services.actionServiceForDoc(this.element);
     this.viewport_ = Services.viewportForDoc(this.element);
 
+    const inputElements = childElementsByTag(this.element, 'input');
+    const textareaElements = childElementsByTag(this.element, 'textarea');
+    userAssert(
+      inputElements.length + textareaElements.length === 1,
+      `${TAG} should contain exactly one <input> or <textarea> child`
+    );
+    this.inputElement_ =
```

Comment:
Yes both elements are mode-agnostic and should be fully supported

Reply:
In that case maybe encapsulate that in a function.

Also can you `querySelectorAll()` for both and simplify? I don't think validation would allow non-children `input/textarea` descendants anyway.

```js
function getSingleInputOrTextareaFrom(element) {
  const possibleElements = element.querySelectorAll('input,textarea');
  userAssert(
    possibleElements.length === 1,
    '%s should contain exactly one <input> or <textarea> descendant %s',
    TAG,
    element
  );
  return /** @type {!HTMLInputElement} */ (possibleElements[0]);
}
```